plugins/search-backend/src/service/router.ts

```diff
@@ -25,10 +25,28 @@ export type RouterOptions = {
   logger: Logger;
 };
 
+const allowedLocationProtocols = ['http:', 'https:'];
+
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
   const { engine, logger } = options;
+
+  const filterResultSet = ({ results, ...resultSet }: SearchResultSet) => ({
+    ...resultSet,
+    results: results.filter(result => {
+      const protocol = new URL(result.document.location, 'https://example.com')
+        .protocol;
+      const isAllowed = allowedLocationProtocols.includes(protocol);
```

Comment:
Hm is the URL fiddling necessary? Could this have been

```ts
const isAllowed = allowedLocationProtocols.some(p => result.document.location.startsWith(p));
```

?

Reply:
I had originally but I wasn't sure if startsWith('/') was robust enough to handle paths.